src/main/kotlin/no/nav/syfo/consumer/narmesteleder/NarmesteLederConsumer.kt

```diff
@@ -1,56 +1,72 @@
 import no.nav.syfo.util.*
 import org.slf4j.LoggerFactory
-import org.springframework.beans.factory.annotation.Autowired
-import org.springframework.beans.factory.annotation.Value
+import org.springframework.beans.factory.annotation.*
 import org.springframework.cache.annotation.Cacheable
 import org.springframework.core.ParameterizedTypeReference
 import org.springframework.http.*
 import org.springframework.stereotype.Component
 import org.springframework.web.client.RestClientResponseException
 import org.springframework.web.client.RestTemplate
-import org.springframework.web.util.UriComponentsBuilder
 
 @Component
 class NarmesteLederConsumer @Autowired constructor(
+    @Value("\${isnarmesteleder.host}") private val isnarmestelederHost: String,
+    @Value("\${isnarmesteleder.id}") private val isnarmestelederId: String,
     @Value("\${syfonarmesteleder.url}") private val syfonarmestelederUrl: String,
+    @Value("\${syfonarmesteleder.id}") private val syfonarmestelederId: String,
     private val azureAdTokenConsumer: AzureAdTokenConsumer,
+    private val azureAdV2TokenConsumer: AzureAdV2TokenConsumer,
     private val metric: Metric,
     private val restTemplate: RestTemplate,
-    @Value("\${syfonarmesteleder.id}") private val syfonarmestelederId: String
+    @Qualifier("restTemplateWithProxy") private val restTemplateWithProxy: RestTemplate,
+    private val contextHolder: OIDCRequestContextHolder
 ) {
+    private val isnarmestelederBaseUrl: String
     private val syfonarmestelederBaseUrl: String
 
     init {
+        this.isnarmestelederBaseUrl = "$isnarmestelederHost/api/v1/narmestelederrelasjon"
         this.syfonarmestelederBaseUrl = "$syfonarmestelederUrl/syfonarmesteleder"
     }
 
-    @Cacheable(value = [CACHENAME_NARMESTELEDER_LEDER], key = "#aktorId + #virksomhetsnummer", condition = "#aktorId != null && #virksomhetsnummer != null")
-    fun narmesteLederRelasjonLeder(aktorId: String, virksomhetsnummer: String): NarmesteLederRelasjon? {
+    @Cacheable(value = [CACHENAME_ISNARMESTELEDER_LEDERE], key = "#innbyggerIdent ", condition = "#innbyggerIdent != null")
+    fun ledereForInnbygger(innbyggerIdent: String): List<NarmesteLederRelasjonDTO> {
         val callId = createCallId()
         try {
-            val response = restTemplate.exchange(
-                getLederUrl(aktorId, virksomhetsnummer),
+            val response = restTemplateWithProxy.exchange(
+                "$isnarmestelederBaseUrl/personident",
                 HttpMethod.GET,
-                entity(callId),
-                NarmestelederResponse::class.java
+                entityWithOboToken(innbyggerIdent),
+                object : ParameterizedTypeReference<List<NarmesteLederRelasjonDTO>>() {}
             )
-            metric.countEvent(CALL_SYFONARMESTELEDER_LEDER_SUCCESS)
 
-            return response.body.narmesteLederRelasjon
+            metric.countEvent(CALL_ISNARMESTELEDER_LEDERE_SUCCESS)
+
+            return response.body
```

Comment:
Får warning i IntelliJ her - er det noe vi bør bry oss med?
`Type mismatch.
Required:
List<NarmesteLederRelasjonDTO>
Found:
List<NarmesteLederRelasjonDTO>?`

Reply:
Hmm, jeg har ikke tatt høyde for at restTemplate kan returnere tom body 🤔